src/storage/storage.service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+    Injectable,
+    InternalServerErrorException,
+    Logger,
+    LoggerService,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+    ReadStream,
+    WriteStream,
+    createReadStream,
+    createWriteStream,
+} from 'fs';
+import { readdir } from 'fs/promises';
+import { join } from 'path';
+import { v4 as uuid } from 'uuid';
+import * as fs from 'fs';
+
+@Injectable()
+export class StorageService {
+    constructor(private readonly configService: ConfigService) {}
+
+    async saveFile(file: Express.Multer.File): Promise<string> {
+        const storage: string | undefined =
+            this.configService.get<string>('STORAGE');
+
+        if (!storage) {
+            throw new Error('Something wrong with env or config service');
+        }
+
+        const fileId: string = uuid();
+        const fileExtension: string = file.mimetype.includes('html')
+            ? 'html'
+            : 'xml';
+        const writeStream: WriteStream = createWriteStream(
+            join(process.cwd(), `${storage}/${fileId}.${fileExtension}`),
+        );
+        writeStream.write(file);
+
+        return fileId;
+    }
+
+    async readFile(name: string): Promise<ReadStream> {
+        const storage: string | undefined =
+            this.configService.get<string>('STORAGE');
+
+        if (!storage) {
+            throw new Error('Something wrong with env or config service');
+        }
+
+        let filePath: string = '';
+        const dir: string = join(process.cwd(), `${storage}`);
+        const files = await readdir(dir);
+
```

Comment:
try catch? Nie wiadomo co złapie błąd jeśli nie będzie dir

Reply:
Generalnie jeżeli nie ma jakiejś logiki w catch to nie trzeba bloku try catch w NestJs, przy zdefini0wanym Gloabl Filter. Nest generalnie odradza stosowania try cacth, bo nie ma za bardzo sensu używania tego. Nest ma wbudowane łapanie wszystkich błędów. Jako, że my mamy global filter co łapie wszystko to tego readdir też złapie i nam programistom wyświeli błąd w Logger w terminalu, a klientowi poleci customowy error 500 - Interal server error